### PR TITLE
SIL: Terminology change: [fragile] => [serialized]

### PR DESCRIPTION
**Pull request by slavapestov**
Also, add a third [serializable] state for functions whose bodies we
*can* serialize, but only do so if they're referenced from another
serialized function.

This will be used for bodies synthesized for imported definitions,
such as init(rawValue:), etc, and various thunks, but for now this
change is NFC.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test

File: include/swift/SIL/SILLinkage.h [hunk 1]
```diff
@@ -86,10 +86,15 @@ enum {
...
-  IsFragile
+/// Related to linkage: flag if a function or global variable is serialized,
+/// either unconditionally, or if referenced from another serialized function.
+enum IsSerialized_t : unsigned char {
+  // Never serialized.
+  IsNotSerialized,
+  // Serialized if referenced from another serialized function.
+  IsSerializable,
+  // Always serialized.
+  IsSerialized
```
Comment (swiftix) [inline include/swift/SIL/SILLinkage.h]: IMHO, `IsSerialized` is not a very good choice for a name. It is too easy to think that something was serialized by the SIL serializer already. I'd prefer names which would use a different verb. Or they could indicate the possibility, e.g. `NotToBeSerialized`, `CanBeSerialized`, `ShouldBeSerialized`....

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: include/swift/SIL/SILLinkage.h (hunk 1 above)
Comment (slavapestov) [inline include/swift/SIL/SILLinkage.h]: It's better than "fragile" no? :-)

Comment (swiftix) [inline include/swift/SIL/SILLinkage.h]: Yes, it is better than "fragile", but not good enough, IMHO ;-)

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test Linux

**Comment by slavapestov** [comment]
@swift-ci Please test Linux

**Comment by slavapestov** [comment]
@swift-ci Please smoke test Linux

**Comment by slavapestov** [comment]
First a flaky SwiftPM test now Foundation, neither related to my change which is NFC. Sigh.

**Comment by slavapestov** [comment]
@swift-ci Please smoke test Linux

**Comment by slavapestov** [comment]
@shahmishal @erg Can one of you guys just click the magic merge button? This patch is just a giant NFC renaming and delaying it for another day will just mean more fixing merge conflicts and rebasing.

**Pull request merged**